### PR TITLE
[JENKINS-33808] Support for Item categorization

### PR DESCRIPTION
**Pull request by recena**
[JENKINS-33808](https://issues.jenkins-ci.org/browse/JENKINS-33808)

downstream of https://github.com/jenkinsci/scm-api-plugin/pull/9

@reviewbybees especially @jglick 

### Screenshot

jenkins-33808

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubBuildStatusNotification.java [hunk 1]
```diff
@@ -95,7 +95,7 @@ private static void createBuildCommitStatus(Run<?,?> build, TaskListener listene
                     Result result = build.getResult();
                     String revisionToNotify = resolveHeadCommit(repo, revision);
                     if (Result.SUCCESS.equals(result)) {
-                        createCommitStatus(repo, revisionToNotify, GHCommitState.SUCCESS, url, Messages.GitHubBuildStatusNotification_CoomitStatus_Good());
+                        createCommitStatus(repo, revisionToNotify, GHCommitState.SUCCESS, url, Messages.GitHubBuildStatusNotification_CommitStatus_Good());
```
Comment (recena) [inline src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubBuildStatusNotification.java]: I know it, *unrelated* but needed.

File: src/main/resources/org/jenkinsci/plugins/github_branch_source/Messages.properties [hunk 2]
```diff
@@ -1,6 +1,9 @@
-GitHubBuildStatusNotification.CoomitStatus.Good=This commit looks good
+GitHubBuildStatusNotification.CommitStatus.Good=This commit looks good
```
Comment (recena) [inline src/main/resources/org/jenkinsci/plugins/github_branch_source/Messages.properties]: I know it, unrelated but needed.

File: src/main/resources/org/jenkinsci/plugins/github_branch_source/GitHubSCMNavigator/newInstanceDetail.jelly [hunk 3]
```diff
@@ -24,6 +24,6 @@ THE SOFTWARE.
 -->
 
 <?jelly escape-by-default='true'?>
```
Comment (jglick) [inline src/main/resources/org/jenkinsci/plugins/github_branch_source/GitHubSCMNavigator/newInstanceDetail.jelly]: Should be able to delete this file.

Comment (recena) [inline src/main/resources/org/jenkinsci/plugins/github_branch_source/GitHubSCMNavigator/newInstanceDetail.jelly]: Sure, [here](https://github.com/jenkinsci/github-branch-source-plugin/pull/37/commits/bde1a8eecdd32f9fb3ecbf661b0ece94bd76f52f)

**Comment by jglick** [comment]
:bee:

**Comment by jglick** [comment]
Need to resolve a merge conflict.

**Comment by recena** [comment]
@jglick Solved the merge conflict.

**Comment by recena** [comment]
@jglick I'm going to cut a release of `scm-api` before to merge.

**Comment by jglick** [comment]
:bee:

**Pull request merged**